Dockerfile

```diff
@@ -5,32 +5,36 @@ LABEL maintainer="CSC Service Desk <servicedesk@csc.fi>"
 # These need to be owned and writable by the root group in OpenShift
 ENV ROOT_GROUP_DIRS='/var/run /var/log/nginx /var/lib/nginx'
 
+ARG repo_org=CSCfi
+ARG repo_name=csc-user-guide
 ARG repo_branch=master
 
-RUN yum -y install epel-release &&\
-    yum -y install nginx &&\
-    yum -y install python38 &&\
-    yum -y install git &&\
-    yum -y install findutils &&\
-    yum clean all
+COPY requirements.txt /tmp
+
+WORKDIR /tmp
+
+RUN dnf -y install epel-release \
+                   nginx \
+                   python3.11 \
+                   python3.11-pip \
+                   git \
+                   findutils &&\
+    dnf clean all &&\
+    pip3 install --use-pep517 --no-cache-dir -r requirements.txt
 
 RUN chgrp -R root ${ROOT_GROUP_DIRS} &&\
     chmod -R g+rwx ${ROOT_GROUP_DIRS}
 
 COPY . /tmp
 
-WORKDIR /tmp
-
-
-RUN git clone --no-checkout https://github.com/CSCfi/csc-user-guide git_folder && \
-    if [ -d ".git" ]; then rm -r .git; fi && \
+RUN if [ ! -d ".git" ]; then \
+    git clone --no-checkout https://github.com/$repo_org/$repo_name git_folder && \
     mv git_folder/.git . && \
     rm -r git_folder && \
     git reset HEAD --hard && \
-    git checkout -f $repo_branch
+    git checkout -f $repo_branch; fi
 
-RUN pip3 install --no-cache-dir -r requirements.txt && \
-    bash scripts/generate_alpha.sh && \
+RUN bash scripts/generate_alpha.sh && \
     bash scripts/generate_by_system.sh && \
     bash scripts/generate_new.sh && \
     bash scripts/generate_glossary.sh && \
```

Comment:
Why not join these two `RUN`s?

If they are joined and the `.git` directory is deleted as a last step, there will be only one layer and it will be smaller.

Reply:
I've now joined them. For getting the missing `.git` directory, I was thinking

```dockerfile
RUN if [ ! -d ".git" ]; then \
    git clone --bare https://github.com/$repo_org/$repo_name .git && \
    git init && \
    git switch --force $repo_branch; fi && \
    bash scripts/generate_alpha.sh && \
    bash scripts/generate_by_system.sh && \
    bash scripts/generate_new.sh && \
    bash scripts/generate_glossary.sh && \
    mkdocs build -d /usr/share/nginx/html
```

I don't know if it's any quicker, but it is more readable.